### ISSUE TITLE
Generate static JSON of example values

### ISSUE THREAD

**Issue by baltpeter** [tweaselORG/tracker-wiki#3]
We ultimately want to have a public database of tracking requests and dynamically list examples of observed values for each data path.

Until we get there, I want to generate a static JSON of such examples values based on the data from my [master's thesis](https://github.com/baltpeter/thesis-mobile-consent-dialogs).

**Comment by baltpeter** [comment]
Updated code that takes this into account:

```ts
import { db, pg } from './db';
import type { AnnotatedResult, Request } from 'trackhar';
import { processRequest } from 'trackhar';
import { writeFile } from 'fs/promises';

type DbRequest = {
    appId: string;
    platform: 'android' | 'ios';
    appVersion: string;

    content?: Buffer;
} & Omit<Request, 'content'>;

(async () => {
    const requests = await db
        .manyOrNone<DbRequest>(
            `select requests.id as "requestId", apps.name as "appId", apps.platform as platform, apps.version as "appVersion", requests.start_time as "startTime", requests.method as method, requests.host as host, requests.path as path, content_raw as content, requests.port as port, requests.scheme as scheme, requests.http_version as "httpVersion", regexp_replace(concat(requests.scheme, '://', requests.host, requests.path), '\\?.+$', '') as "endpointUrl"
        from apps
          join runs r on apps.id = r.app
          join requests on r.id = requests.run;`
        )
        .then((reqs) => reqs.map((r) => ({ ...r, content: r.content?.toString('binary') })));

    const processedRequests = requests
        .map((r) => {
            try {
                return processRequest(r);
            } catch (err) {
                console.error(err, r);
                throw err;
            }
        })
        .filter((r): r is AnnotatedResult => !!r);

    const values = processedRequests.reduce<Record<string, Set<any>>>((acc, cur) => {
        for (const r of cur) {
            // https://github.com/tweaselORG/tracker-wiki/issues/3#issuecomment-1505548347
            const key = `${r.adapter}/${r.property.toLowerCase()}/${r.context}/${r.path}`;
            if (!acc[key]) acc[key] = new Set();

            acc[key].add(['string', 'number'].includes(typeof r.value) ? r.value : JSON.stringify(r.value));
        }

        return acc;
    }, {});

    const json = JSON.stringify(values, (k, v) => (v instanceof Set ? [...v] : v), 4);
    await writeFile('adapterExamples.json', json);

    pg.end();
})();

process.on('unhandledRejection', (err) => {
    console.error('An unhandled promise rejection occurred:', err);
    pg.end();
    process.exit(1);
});
```

**Comment by baltpeter** [comment]
Fixed with #1.
